File: Week1/credit.c
#include <stdio.h>
#include <stdlib.h>
#include <cs50.h>

int main(){
    long card_number = get_long("Number: ");
    int sum = 0;
    long copy = card_number; // not changing original cardnumber.
    int count = 0;
    while(copy){
        copy /= 10;
        count ++ ;
    }
    if (count < 13) printf("INVALID\n");
    else{
    copy = card_number; //not wanting to change original card number
    do{
        copy /= 10;
        if((copy%10)*2 < 10) sum += (copy%10)*2;
        else{
            sum += ((copy%10)*2)%10 + 1;
        }
        copy/= 10;
    }while(copy);
    copy = card_number; //not wanting to chagnge original card_number
    int newsum = 0;
    do{
        newsum += (copy % 10);
        copy /= 10;
        copy /= 10;
    }while(copy);
    newsum += sum;

    if(newsum%10 == 0){
        bool printed = false;
        while(card_number && !printed){
            switch(card_number){ //You can use if-else in place of switch
                case 4:
                    printf("VISA\n");
                    printed = true;
                    break;
                case 34:
                case 37:
                    printf("AMEX\n");
                    printed = true;
                    break;
                case 51:
                case 52:
                case 53:
                case 54:
                case 55:
                    printf("MASTERCARD\n");
                    printed = true;
                    break;
                default:
                    card_number /= 10;
            }
        }
        if (!printed) printf("INVALID\n");
    }
    else printf("INVALID\n");
    }
}

File: Week6/sentimental-credit/credit.py
def main():
    # getting user card number
    card_number = int(input("Number: "))

    copy = card_number

    # length of card_number
    count = 0
    while (copy):
        copy //= 10
        count += 1

    # validating length of card number
    if (count < 13):
        print("INVALID")
        return 0

    copy = card_number

    # alternate sum
    sum = 0
    while (copy):
        copy //= 10
        if ((copy % 10) * 2 < 10):
            sum += (copy % 10) * 2
        else:
            sum += ((copy % 10) * 2) % 10 + 1
        copy //= 10

    copy = card_number

    # final sum
    newsum = 0
    while (copy):
        newsum += copy % 10
        copy //= 10
        copy //= 10

    newsum += sum

    # determining card type
    if (newsum % 10 == 0):
        number = int(str(card_number)[0:2])
        if (number >= 40 and number <= 49):
            print("VISA")
        elif (number == 34 or number == 37):
            print("AMEX")
        elif (number >= 51 and number <= 55):
            print("MASTERCARD")
        else:
            print("INVALID")
    else:
        print("INVALID")


# execute the main function
if __name__ == "__main__":
    main()
